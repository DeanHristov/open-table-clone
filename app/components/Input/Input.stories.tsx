import * as React from "react";
import { Meta, Story } from "@storybook/react";
import Input, { IInputProps } from "./index";

export default {
  title: "Components/Input",
  component: Input,
} as Meta;

export const Default: Story<IInputProps> = (args) => (
  <Input placeholder={"Type location..."} {...args} />
);

Default.args = {};
