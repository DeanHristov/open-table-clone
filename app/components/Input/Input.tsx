import { FC, InputHTMLAttributes } from "react";
import { MagnifyingGlassIcon } from "@heroicons/react/24/solid";

export interface IInputProps extends InputHTMLAttributes<HTMLInputElement> {}

const Input: FC<IInputProps> = (props) => {
  return (
    <div className="min-w-[290px] px-2 md:px-0 w-full relative">
      <input
        className="border border-gray-200 rounded md:mr-3 pl-10 py-2 h-12 w-full"
        {...props}
      />
      <MagnifyingGlassIcon
        width={24}
        className="absolute top-1/4 left-4 text-black"
      />
    </div>
  );
};

export default Input;
