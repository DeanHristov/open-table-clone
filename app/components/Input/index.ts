import Input, { IInputProps } from "./Input";

export type { IInputProps };

export default Input;
