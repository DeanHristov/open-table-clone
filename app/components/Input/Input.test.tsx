import { render } from "@testing-library/react";
import Input from "./Input";

describe("Component: <Input {...} />", () => {
  it("Should be able to render default state", () => {
    const { container } = render(<Input placeholder="search location" />);

    expect(container).toMatchSnapshot();
  });
});
