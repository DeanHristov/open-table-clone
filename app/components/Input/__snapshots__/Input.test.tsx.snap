// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Input {...} /> Should be able to render default state 1`] = `
<div>
  <div
    class="min-w-[290px] px-2 md:px-0 w-full relative"
  >
    <input
      class="border border-gray-200 rounded md:mr-3 pl-10 py-2 h-12 w-full"
      placeholder="search location"
    />
    <svg
      aria-hidden="true"
      class="absolute top-1/4 left-4 text-black"
      fill="currentColor"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10.5 3.75a6.75 6.75 0 100 13.5 6.75 6.75 0 000-13.5zM2.25 10.5a8.25 8.25 0 1114.59 5.28l4.69 4.69a.75.75 0 11-1.06 1.06l-4.69-4.69A8.25 8.25 0 012.25 10.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
